flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeTypeSerializerConfigSnapshot.java

```diff
@@ -110,4 +123,20 @@ public boolean equals(Object obj) {
 	public int hashCode() {
 		return nestedSerializersAndConfigs.hashCode();
 	}
+
+	private Map<TypeSerializer<?>, Integer> buildSerializerIndices() {
```

Comment:
While the idea makes sense, I am not sure if we can rely on all serializer having a correct implementation of equals and hash code that will do what we want. We might be better of with collection of `serializer -> index` pairs and using equality for deduplication, but even there equals might be tricky. While it is ok to accidentally write serializer twice, it should never happen that a serializer gets de-duplicated by accident.